### PR TITLE
fix `from_str` serializer

### PR DESCRIPTION
**Pull request by noot**
* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Added entry in `.changelog/`

This fixes the `from_str` serialization. I discovered this while trying to serialize and deserialize the response type for `broadcast_tx_commit` while making a mock tendermint server; the `gas_wanted` field and a few others were serialized as integers, but were decoded as a string, causing an error `invalid type: integer `0`, expected a string at line 1 column 98`. `Display` formats the integer without quotes, but we wanted with quotes.

Before:
```
"{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":{\"check_tx\":{\"code\":0,\"data\":[],\"log\":\"\",\"info\":\"\",\"gas_wanted\":0,\"gas_used\":0,\"events\":[],\"codespace\":\"\",\"sender\":\"\",\"priority\":0,\"mempool_error\":\"\"},\"deliver_tx\":{\"code\":0,\"data\":[],\"log\":\"\",\"info\":\"\",\"gas_wanted\":0,\"gas_used\":0,\"events\":[],\"codespace\":\"\"},\"hash\":\"\",\"height\":\"1\"},\"error\":null}"
```

After:
```
"{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":{\"check_tx\":{\"code\":0,\"data\":null,\"log\":\"\",\"info\":\"\",\"gas_wanted\":\"0\",\"gas_used\":\"0\",\"events\":[],\"codespace\":\"\",\"sender\":\"\",\"priority\":\"0\",\"mempool_error\":\"\"},\"deliver_tx\":{\"code\":0,\"data\":null,\"log\":\"\",\"info\":\"\",\"gas_wanted\":\"0\",\"gas_used\":\"0\",\"events\":[],\"codespace\":\"\"},\"hash\":\"0000000000000000000000000000000000000000000000000000000000000000\",\"height\":\"1\"},\"error\":null}"
```

Also made `rpc::Wrapper::new_with_id` public because I'm using it externally.

### CODE REVIEW

File: rpc/src/response.rs [hunk 1]
```diff
@@ -69,7 +69,6 @@ impl<R> Wrapper<R> {
         }
     }
 
-    #[cfg(test)]
```
Comment (thanethomson) [inline rpc/src/response.rs:72]: IIRC this method was only ever supposed to be used in tests. Why the change?

Comment (noot) [reply rpc/src/response.rs:72]: I wanted to use this type externally and there's no other way to construct it afaik. I can revert this and just make my own `Wrapper` type if you'd prefer

**Review by thanethomson — APPROVED** [review]

**Comment by thanethomson** [comment]
Thanks @noot!
